src/type-translator.ts

```diff
@@ -125,10 +125,10 @@ export function symbolToDebugString(sym: ts.Symbol): string {
 /** TypeTranslator translates TypeScript types to Closure types. */
 export class TypeTranslator {
   /**
-   * A list of types we've encountered while emitting; used to avoid getting stuck in recursive
-   * types.
+   * A list of type literals we've encountered while emitting; used to avoid getting stuck in
+   * recursive types.
    */
-  private readonly seenTypes: ts.Type[] = [];
+  private readonly seenTypeLiterals: ts.Type[] = [];
```

Comment:
You can use a Set<> here.

Reply:
I stumbled over that, and was wondering if you used an array on purpose given these are very likely never more than one or two types.